### PR TITLE
[efr32] add radio.c debug counters for efr32mg12 and efr32mg21

### PR DESCRIPTION
**Pull request by marvensilabs**
Debug counters added in radio.c for mg12 and mg21.

### CODE REVIEW

File: examples/platforms/efr32mg12/brd4166a/board_config.h [hunk 1]
```diff
@@ -35,6 +35,8 @@
 #ifndef __BOARD_CONFIG_H__
 #define __BOARD_CONFIG_H__
 
-#define RADIO_CONFIG_2P4GHZ_OQPSK_SUPPORT 1 ///< Dev board suppports OQPSK modulation in 2.4GHz band.
+#define RADIO_CONFIG_2P4GHZ_OQPSK_SUPPOR 1 ///< Dev board suppports OQPSK modulation in 2.4GHz band.
```
Comment (jwhui) [inline examples/platforms/efr32mg12/brd4166a/board_config.h]: typo: `RADIO_CONFIG_2P4GHZ_OQPSK_SUPPOR` -> `RADIO_CONFIG_2P4GHZ_OQPSK_SUPPORT`

File: examples/platforms/efr32mg12/platform-band.h [hunk 2]
```diff
@@ -43,6 +43,26 @@
 
 #define RAIL_TX_FIFO_SIZE (OT_RADIO_FRAME_MAX_SIZE + 1)
 
+typedef struct efr32RadioCounters
+{
+    uint64_t mRailPlatTxTriggered;
+    uint64_t mRailPlatRadioReceiveDoneCbCount;
+    uint64_t mRailPlatRadioEnergyScanDoneCbCount;
```
Comment (jwhui) [inline examples/platforms/efr32mg12/platform-band.h]: Is this used somewhere?

File: examples/platforms/efr32mg12/platform-band.h [hunk 3]
```diff
@@ -43,6 +43,26 @@
...
+    uint64_t mRailPlatRadioTxDoneCbCount;
+    uint64_t mRailTxStarted;
+    uint64_t mRailTxStartFailed;
+    uint64_t mRailEventConfigScheduled;
+    uint64_t mRailEventConfigUnScheduled;
+    uint64_t mRailEventPacketSent;
+    uint64_t mRailEventChannelBusy;
+    uint64_t mRailEventEnergyScanCompleted;
```
Comment (jwhui) [inline examples/platforms/efr32mg12/platform-band.h]: Is this used somewhere?

File: examples/platforms/efr32mg12/platform-band.h [hunk 4]
```diff
@@ -43,6 +43,26 @@
...
+    uint64_t mRailEventCalNeeded;
```
Comment (jwhui) [inline examples/platforms/efr32mg12/platform-band.h]: Is this used somewhere?

File: examples/platforms/efr32mg12/brd4166a/board_config.h (hunk 1 above)
Comment (marvensilabs) [reply examples/platforms/efr32mg12/brd4166a/board_config.h]: Fixed

File: examples/platforms/efr32mg12/platform-band.h (hunk 2 above)
Comment (marvensilabs) [reply examples/platforms/efr32mg12/platform-band.h]: Was missing, using it in latest commit

File: examples/platforms/efr32mg12/platform-band.h (hunk 4 above)
Comment (marvensilabs) [reply examples/platforms/efr32mg12/platform-band.h]: Was missing, using it in latest commit

File: examples/platforms/efr32mg12/platform-band.h (hunk 3 above)
Comment (marvensilabs) [reply examples/platforms/efr32mg12/platform-band.h]: Was missing, using it in latest commit

**Comment by marvensilabs** [comment]
Additionally added Silicon Laboratories, Inc. to AUTHORS file.

**Pull request merged**